### PR TITLE
[MOD-2351] Support Android

### PR DESCRIPTION
**Pull request by garymathews**
- Implement `ti.worker` for Android

[JIRA Ticket](https://jira.appcelerator.org/browse/MOD-2351)

### CODE REVIEW

**Comment by lokeshchdhry** [comment]
@garymathews ,
Running the example I get this error:
```
[ERROR] :  [Pixel] TiExceptionHandler: (main) [5,105]   worker.postMessage(event.data.msg);
[ERROR] :  [Pixel] TiExceptionHandler:                                 ^
[ERROR] :  [Pixel] TiExceptionHandler: TypeError: Cannot read property 'msg' of null
[ERROR] :  [Pixel] TiExceptionHandler:     at TiWorker.eval (eval at <anonymous> (<anonymous>:1:236), <anonymous>:6:33)
[ERROR] :  [Pixel] TiExceptionHandler:     at TiWorker.value (ti:/events.js:50:21)
[ERROR] :  [Pixel] TiExceptionHandler:     at TiWorker.value (ti:/events.js:107:20)
[ERROR] :  [Pixel] TiExceptionHandler:     at TiWorker.eval (eval at <anonymous> (<anonymous>:1:236), <anonymous>:6:10)
[ERROR] :  [Pixel] TiExceptionHandler:     at TiWorker.value (ti:/events.js:50:21)
[ERROR] :  [Pixel] TiExceptionHandler:     at TiWorker.value (ti:/events.js:107:20)
[ERROR] :  [Pixel] TiExceptionHandler:     at TiWorker.eval (eval at <anonymous> (<anonymous>:1:236), <anonymous>:6:10)
[ERROR] :  [Pixel] TiExceptionHandler:     at TiWorker.value (ti:/events.js:50:21)
[ERROR] :  [Pixel] TiExceptionHandler:     at TiWorker.value (ti:/events.js:107:20)
[ERROR] :  [Pixel] TiExceptionHandler:     at /app.js:18:6
[ERROR] :  [Pixel] TiExceptionHandler:
[ERROR] :  [Pixel] TiExceptionHandler:     org.appcelerator.kroll.runtime.v8.V8Object.nativeFireEvent(Native Method)
[ERROR] :  [Pixel] TiExceptionHandler:     org.appcelerator.kroll.runtime.v8.V8Object.fireEvent(V8Object.java:63)
[ERROR] :  [Pixel] TiExceptionHandler:     org.appcelerator.kroll.KrollProxy.doFireEvent(KrollProxy.java:979)
[ERROR] :  [Pixel] TiExceptionHandler:     org.appcelerator.kroll.KrollProxy.fireSyncEvent(KrollProxy.java:837)
[ERROR] :  [Pixel] TiExceptionHandler:     ti.worker.TiWorkerProxy.postMessage(TiWorkerProxy.java:58)
[ERROR] :  [Pixel] TiExceptionHandler:     org.appcelerator.kroll.runtime.v8.V8Object.nativeFireEvent(Native Method)
[ERROR] :  [Pixel] TiExceptionHandler:     org.appcelerator.kroll.runtime.v8.V8Object.fireEvent(V8Object.java:63)
[ERROR] :  [Pixel] TiExceptionHandler:     org.appcelerator.kroll.KrollProxy.doFireEvent(KrollProxy.java:979)
[ERROR] :  [Pixel] TiExceptionHandler:     org.appcelerator.kroll.KrollProxy.fireSyncEvent(KrollProxy.java:837)
[ERROR] :  [Pixel] TiExceptionHandler:     ti.worker.TiWorkerProxy.postMessage(TiWorkerProxy.java:58)
```

**Comment by lokeshchdhry** [comment]
FR Passed.

Studio Ver:     5.1.2.201812191831
SDK Ver:        8.0.0 local build
OS Ver:         10.14
Xcode Ver:      Xcode 10.1
Appc NPM:       4.2.13
Appc CLI:       7.0.9
Daemon Ver:     1.1.3
Ti CLI Ver:     5.1.1
Alloy Ver:      1.13.4
Node Ver:       8.9.1
NPM Ver:        5.5.1
Java Ver:       10.0.2
Devices:        ⇨ google Nexus 5 (Android 6.0.1)
                ⇨ google Pixel (Android 9)
Module: ti.worker (1.0.0)

File: android/src/ti/worker/TiWorkerProxy.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	public void terminate() {
+		fireSyncEvent("terminated", null);
+	}
+
+	@Kroll.method
+	public void postMessage(Object message) {
+		if (message != null) {
+			final KrollDict data = new KrollDict();
+			data.put("data", message);
+			fireSyncEvent("message", data);
```
Comment (drauggres) [inline android/src/ti/worker/TiWorkerProxy.java]: This made the module to work like "echo" by itself.
Every call of `workerInstance.postMessage()` will produce `message` event on it (from both sides: main and worker scriprts).

To test it use app.js from [Readme](/appcelerator-modules/ti.worker/README.md) and **empty** `echo.js`

**Comment by drauggres** [comment]
Also [`postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage) method of worker instance (in main script) and [`postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/DedicatedWorkerGlobalScope/postMessage) method inside woker script are two completely different methods.

If you'll change `echo.js` from [Readme](/appcelerator-modules/ti.worker/README.md) to:

  ```javascript
// subscribe to events send with postMessage
worker.addEventListener('message',function(event){

	// send data back to any subscribers
	// pull data from the event from the data property
	worker.postMessage({msg: event.data.msg});  // <--CHANGE HERE
});
```
you will get infinite loop and eventually exceed maximum call stack size

**Pull request merged**